src/components/modal/index.js

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect } from 'react';
+import classnames from 'classnames';
+import CloseIcon from '../icons/close';
+import IconButton from '../../components/icon-button';
+import './modal.css';
+
+/**
+ * Generic Kedro Modal
+ */
+const Modal = ({ title, onClose, visible, message, children }) => {
+  const handleKeyDown = (event) => {
+    if (event.keyCode === 27) {
```

Comment:
great one for adding this in ( good spot @tynandebold ) . With the introduction of the keycode events into our codebase, we should set up a `handleKeyEvents` function in utils to handle all the key events that are used in our app, especially now that we are migrating kedro-UI into our codebase - the set up of the function can reference the structure as set out in Kedro-UI [here](https://github.com/quantumblacklabs/kedro-ui/blob/master/src/utils/key-events.js). 

Setting this up will also make the future migration of other components from Kedro-ui easier. 😄

Reply:
hey @studioswong  -- there's a future ticket in the backlog to do that https://jira.quantumblack.com/browse/KED-2985 
I was thinking of doing the migration to utils folder as part of that ticket.

Reply:
I see the utility migration being in the pipeline as a later ticket - that said, I would rather we take the migration of utility functions as part of the end-to-end migration of the component as a complete feature rather than as a separate ticket, particularly given that we only want to migrate the parts within the utility function that we use into the codebase, and doing it seperately will easily follow the trap of migrating unnecessary code ( while also extremely easy to forget to come back to refactor other part of the codebase, especially if the utility migration ticket is picked up by someone else.)

in other words, my recommendation is:
1) set up a `handleKeyEvents` utility function that only tackles the handleKeyDown event as needed here as part of the complete work for this ticket
2) No need to migrate the `handleKeyFunction` later on as it is set up already.

Hope this is clear?